mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_jwt.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package resourcemanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+	"google.golang.org/api/iamcredentials/v1"
+)
+
+var _ ephemeral.EphemeralResource = &googleEphemeralServiceAccountJwt{}
+
+func GoogleEphemeralServiceAccountJwt() ephemeral.EphemeralResource {
+	return &googleEphemeralServiceAccountJwt{}
+}
+
+type googleEphemeralServiceAccountJwt struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+func (p *googleEphemeralServiceAccountJwt) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account_jwt"
+}
+
+type ephemeralServiceAccountJwtModel struct {
+	Payload              types.String `tfsdk:"payload"`
+	ExpiresIn            types.Int64  `tfsdk:"expires_in"`
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	Jwt                  types.String `tfsdk:"jwt"`
+}
+
+func (p *googleEphemeralServiceAccountJwt) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Produces an arbitrary self-signed JWT for service accounts.",
+		Attributes: map[string]schema.Attribute{
+			"payload": schema.StringAttribute{
+				Required:    true,
+				Description: `A JSON-encoded JWT claims set that will be included in the signed JWT.`,
+			},
+			"expires_in": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Number of seconds until the JWT expires. If set and non-zero an `exp` claim will be added to the payload derived from the current timestamp plus expires_in seconds.",
+			},
+			"target_service_account": schema.StringAttribute{
+				Description: "The email of the service account that will sign the JWT.",
+				Required:    true,
+			},
```

Comment:
Please add a Validator to this field, similar to those used in your other PRs. The original data source has validation on the field: https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/resourcemanager/data_source_google_service_account_jwt.go